annotations/component-annotations/src/main/java/io/ap4k/component/handler/ComponentHandler.java

```diff
@@ -46,23 +58,38 @@ public int order() {
   }
 
   @Override
-  public void handle(CompositeConfig config) {
+  public void handle(ComponentConfig config) {
     if (Strings.isNullOrEmpty(resources.getName())) {
       resources.setName(config.getName());
     }
     resources.addCustom(ResourceGroup.NAME, createComponent(config));
+    addVisitors(config);
+
   }
 
   @Override
   public boolean canHandle(Class<? extends Configuration> type) {
-    return type.equals(CompositeConfig.class) ||
-      type.equals(EditableCompositeConfig.class);
+    return type.equals(ComponentConfig.class) ||
+      type.equals(EditableComponentConfig.class);
   }
 
-  private void addVisitors(CompositeConfig config) {
+  private void addVisitors(ComponentConfig config) {
     String type = config.getAttribute(RUNTIME_TYPE);
     String version = config.getAttribute(RUNTIME_VERSION);
 
+    FileRepositoryBuilder builder = new FileRepositoryBuilder();
```

Comment:
This is awesome! In fact its to awesome to have it limited to the component.
Please create and ScmInfo object with `uri`, `branch` and `commit sha` and have as a property of the Project object. Then we can use it from everywehere.